### PR TITLE
(style): initial commit formatting using black

### PR DESCRIPTION
**Pull request by za**
```
➜  URLExtract git:(master) black urlextract

reformatted urlextract/__init__.py
reformatted urlextract/cachefile.py
reformatted urlextract/urlextract_core.py
All done! ✨ 🍰 ✨
3 files reformatted.
```

### CODE REVIEW

**Review by za — COMMENTED** [review]
I've manually edited it.

File: urlextract/urlextract_core.py [hunk 1]
```diff
@@ -51,33 +51,33 @@ class URLExtract(CacheFile):
...
-        r"^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])$")
+    _hostname_re = re.compile(r"^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])$")
 
     # list of enclosure of URL that should be removed
     _enclosure = {
         ("(", ")"),
         ("{", "}"),
         ("[", "]"),
-        ("\"", "\""),
+        ('"', '"'),
```
[the remark this answers is not in the document]
Comment (za) [reply urlextract/urlextract_core.py]: I manually edited. I think we treat `\` as escape character which requires the character after it.

**Comment by za** [comment]
@lipoja You may double check now. I've manually edited the changes related with `\` part.

**Comment by lipoja** [comment]
@za Hi thanks on formatting the code.
However I would like to suggest to add a test to ensure that any feature commit is formatted with Black as well. 
It means to add Black to `tox.ini` so tests will check if the code is formatted properly or not.

**Comment by za** [comment]
@lipoja I've updated the PR. Please check.

And BTW, Black requires Python 3.6

> Black can be installed by running pip install black. It requires Python 3.6.2+ to run.

https://github.com/psf/black#installation

It failed on Python 3.5
```
Collecting dnspython
  Downloading dnspython-1.16.0-py2.py3-none-any.whl (188 kB)
ERROR: Could not find a version that satisfies the requirement black
ERROR: No matching distribution found for black
Error: Process completed with exit code 1.
```
https://github.com/lipoja/URLExtract/runs/4175554130?check_suite_focus=true

**Comment by lipoja** [comment]
> @lipoja I've updated the PR. Please check.
> 
> And BTW, Black requires Python 3.6
> 
> > Black can be installed by running pip install black. It requires Python 3.6.2+ to run.
> 
> https://github.com/psf/black#installation
> 
> It failed on Python 3.5
> 
> ```
> Collecting dnspython
>   Downloading dnspython-1.16.0-py2.py3-none-any.whl (188 kB)
> ERROR: Could not find a version that satisfies the requirement black
> ERROR: No matching distribution found for black
> Error: Process completed with exit code 1.
> ```
> 
> https://github.com/lipoja/URLExtract/runs/4175554130?check_suite_focus=true

Please, make Black as a separate test in `tox.ini`. We do not have to run it in every python environment.

File: tox.ini [hunk 2]
```diff
@@ -10,3 +10,4 @@ deps =
 commands =
   pytest --ignore setup.py {posargs}
   pytest urlextract {posargs: --doctest-modules}
+  black urlextract --skip-string-normalization
```
Comment (lipoja) [inline tox.ini:13]: Let's move this to `[testenv:black]`

Comment (lipoja) [reply tox.ini:13]: And also lets move  `--skip-string-normalization` to `pyproject.toml` file to `[tool.black]`. So we can set more things for Black in the future.

**Comment by za** [comment]
> Please, make Black as a separate test in tox.ini. We do not have to run it in every python environment.

Hi @lipoja can you check? I've made new commit. Sorry for not updating the PR for few days.

I don't have `python3.6` running on my machine. So when I run `tox`, I got this:

```
black create: /home/za/git/URLExtract/.tox/black
SKIPPED: InterpreterNotFound: python3.6
______________________________________ summary _______________________________________
  py-nocache: commands succeeded
  py-cache: commands succeeded
SKIPPED:  black: InterpreterNotFound: python3.6
  congratulations :)
```

 I have `python3.8` running, after I replaced it with `python3.8`, it's OK:
```
black create: /home/za/git/URLExtract/.tox/black
black installdeps: black
black inst: /home/za/git/URLExtract/.tox/.tmp/package/1/urlextract-1.4.0.zip
black installed: black==21.11b0,click==8.0.3,filelock==3.4.0,idna==3.3,mypy-extensions==0.4.3,pathspec==0.9.0,platformdirs==2.4.0,regex==2021.11.10,tomli==1.2.2,typing_extensions==4.0.0,uritools==3.0.2,urlextract @ file:///home/za/git/URLExtract/.tox/.tmp/package/1/urlextract-1.4.0.zip
black run-test-pre: PYTHONHASHSEED='1420943709'
black run-test: commands[0] | black urlextract --skip-string-normalization
All done! ✨ 🍰 ✨
3 files left unchanged.
```

File: tox.ini (hunk 2 above)
Comment (za) [reply tox.ini:21]: Please check the updates @lipoja

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+[tool.black]
+line-length = 88
+target_version = ["py36", "py37", "py38"]
```
Comment (lipoja) [inline pyproject.toml:3]: Maybe we can have here also 3.9 an 3.10 since they are supported (see `setup.py`)

File: requirements.txt [hunk 4]
```diff
@@ -5,3 +5,4 @@ platformdirs
 bump2version
 filelock
 dnspython
+black
```
Comment (lipoja) [inline requirements.txt:8]: Let's try this `black;python_version>="3.6"` so it won't fail tests in py35.

File: tox.ini (hunk 2 above)
Comment (lipoja) [reply tox.ini:21]: great! Thanks, could you also add `tests` folder as well?

**Review by lipoja — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 2 above)
Comment (za) [reply tox.ini:21]: @lipoja OK, added!

File: requirements.txt (hunk 4 above)
Comment (za) [reply requirements.txt]: @lipoja it works!

**Comment by za** [comment]
Hi @lipoja please check. I've addressed the feedback.

**Comment by lipoja** [comment]
@za Hi, thank you for your time working on this.

File: tox.ini [hunk 5]
```diff
@@ -10,3 +12,11 @@ deps =
...
   pytest --ignore setup.py {posargs}
   pytest urlextract {posargs: --doctest-modules}
+
+[testenv:black]
+basepython = python3.6
+deps =
+  black
+commands =
+  black urlextract --skip-string-normalization
+  black tests --skip-string-normalization
```
Comment (lipoja) [inline tox.ini:22]: I have just small thing. I checked the tests for black and I saw `1 file reformatted, 2 files left unchanged.` Could you just add flag to both black commands that it only checks the file format but does not modify it (should be something like `--check`).

**Review by lipoja — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 5 above)
Comment (za) [reply tox.ini:22]: ok @lipoja

Comment (za) [reply tox.ini]: @lipoja added! Does this mean, we should revert the black-formatted files in `urlextract` and `tests` folder?

**Comment by za** [comment]
> @za Hi, thank you for your time working on this.

No worries, @lipoja ;-)

**Comment by za** [comment]
Hi @lipoja , I see that Black does not support Python 3.10 yet:

```
(.env) ➜  URLExtract git:(black-formatted) black --check --skip-string-normalization tests
Usage: black [OPTIONS] SRC ...
Try 'black -h' for help.

Error: Invalid value for '-t' / '--target-version': 'py310' is not one of 'py27', 'py33', 'py34', 'py35', 'py36', 'py37', 'py38', 'py39'.
```

I'll update `pyproject.toml`.

**Comment by za** [comment]
Hi @lipoja I am having issue with this CI process. I might need your help on this.

File: tox.ini (hunk 5 above)
Comment (lipoja) [reply tox.ini]: No, we should keep them re-formatted in this PR otherwise the CI will fail.

**Comment by lipoja** [comment]
> Hi @lipoja I am having issue with this CI process. I might need your help on this.

It says that one file is not formatted correctly (`would reformat urlextract/urlextract_core.py`) 
Could you run black on the code again and commit the changed file. 
After that it should pass CI tests.

**Comment by za** [comment]
> > Hi @lipoja I am having issue with this CI process. I might need your help on this.
> 
> It says that one file is not formatted correctly (`would reformat urlextract/urlextract_core.py`) Could you run black on the code again and commit the changed file. After that it should pass CI tests.

OK, I added the `,` again @lipoja . You can check my commit message why I am confused https://github.com/lipoja/URLExtract/pull/105/commits/ea21ee769246f69e9b95c7cc1e446be72e602841

**Comment by lipoja** [comment]
> > > Hi @lipoja I am having issue with this CI process. I might need your help on this.
> > 
> > 
> > It says that one file is not formatted correctly (`would reformat urlextract/urlextract_core.py`) Could you run black on the code again and commit the changed file. After that it should pass CI tests.
> 
> OK, I added the `,` again @lipoja . You can check my commit message why I am confused [ea21ee7](https://github.com/lipoja/URLExtract/commit/ea21ee769246f69e9b95c7cc1e446be72e602841)

When I remove the comma ',' it is passing tests locally for me. Could you remove the comma, run black again and then push the changes. And lets see what CI tells us. Thanks!

**Comment by za** [comment]
Hi @lipoja this is the tox result:

```
black inst-nodeps: /home/za/git/URLExtract/.tox/.tmp/package/1/urlextract-1.4.0.zip
black installed: black==21.11b1,click==8.0.3,filelock==3.4.0,idna==3.3,mypy-extensions==0.4.3,pathspec==0.9.0,platformdirs==2.4.0,regex==2021.11.10,tomli==1.2.2,typing_extensions==4.0.0,uritools==3.0.2,urlextract @ file:///home/za/git/URLExtract/.tox/.tmp/package/1/urlextract-1.4.0.zip
black run-test-pre: PYTHONHASHSEED='435205281'
black run-test: commands[0] | black urlextract --check --skip-string-normalization
would reformat urlextract/urlextract_core.py
Oh no! 💥 💔 💥
1 file would be reformatted, 2 files would be left unchanged.
ERROR: InvocationError for command /home/za/git/URLExtract/.tox/black/bin/black urlextract --check --skip-string-normalization (exited with code 1)
__________________________________________________________________________________ summary __________________________________________________________________________________
  py-nocache: commands succeeded
  py-cache: commands succeeded
ERROR:   black: commands failed
```

And this is the Black result:
```
(.env) ➜  URLExtract git:(black-formatted) black urlextract
would reformat urlextract/urlextract_core.py
Oh no! 💥 💔 💥
1 file would be reformatted, 2 files would be left unchanged.
```

**Comment by lipoja** [comment]
@za Oh I see one issue ... we can not set "check" for black by default. We have to just use `--check` only when running black in tox. Otherwise it will not format your files locally. 

Could you please remove `check = true` from `pyproject.toml`, run black locally and push changes... and could you also include `setup.py` to formatting and checking (tox). I forget that before.

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+[tool.black]
+line-length = 88
+target_version = ["py36", "py37", "py38", "py39"]
+skip-string-normalization = true
+check = true
```
Comment (lipoja) [inline pyproject.toml:5]: This should be removed otherwise black does not format code locally.

**Review by lipoja — CHANGES_REQUESTED** [review]

File: urlextract/urlextract_core.py [hunk 7]
```diff
@@ -66,18 +66,18 @@ class URLExtract(CacheFile):
...
-            cache_dns=True,
-            extract_localhost=True,
-            limit=DEFAULT_LIMIT,
-            **kwargs
+        self,
+        extract_email=False,
+        cache_dns=True,
+        extract_localhost=True,
+        limit=DEFAULT_LIMIT,
+        **kwargs
```
Comment (lipoja) [inline urlextract/urlextract_core.py:80]: @za is the comma added here automatically by black? Or what happen?

**Comment by za** [comment]
@lipoja can you check again? Now the `tox` also failed.

File: urlextract/urlextract_core.py (hunk 7 above)
Comment (za) [reply urlextract/urlextract_core.py:80]: @lipoja Yes, Black (suggested) to have comma. Previously it wasn't. It all started with this commit: https://github.com/lipoja/URLExtract/pull/105/commits/866ff39db2e07b922c4e8c075b16c171e6ab1e69 (see the commit message for detail).

**Comment by za** [comment]
Hi @lipoja once you have time, please check the PR. Need your advise/input :-)

**Comment by lipoja** [comment]
Thanks for this PR! I will deal with those errors.

**Comment by lipoja** [comment]
Thanks for this PR! I will deal with those errors.

**Comment by za** [comment]
Wow, thanks for merging it @lipoja I thought we should've done some work before merging this PR. 

Let me know how will you handle and tackle the errors.

**Pull request merged**